### PR TITLE
Add runtime API for return remote management status

### LINKED ISSUE #42712: [Improvement]: Expose remote management build option as runtime API
### Description

$subject

### Describe your problem(s)

This is needed to identify whether or not to register the Ballerina node. 

### Describe your solution(s)

In `Repository` interface
```java
    public boolean isRemoteEnabled();
```

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject 

Fixes #42712 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/RepositoryImpl.java [hunk 1]
```diff
@@ -72,6 +72,11 @@ public Node getNode() {
                 System.getProperty("os.version"));
     }
 
+    @Override
+    public boolean isRemoteEnabled() {
+        return isRemoteEnabled;
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/RepositoryImpl.java:77]: Any reason why it is not covered?

Comment (HindujaB) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/RepositoryImpl.java:77]: This is called from the unit test `RuntimeAPITest` and the interface method is covered. The unit test I added is failing if I remove/change the line.

Comment (HindujaB) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/RepositoryImpl.java:77]: Let me analyze why the codecov fails.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/ConfigMethodGen.java [hunk 2]
```diff
@@ -144,11 +154,27 @@ private void generateConfigInit(ClassWriter cw, Set<PackageID> imprtMods, Packag
             generateInvokeConfigureInit(mv, id);
         }
         generateInvokeConfiguration(mv, packageID);
+        generateBallerinaRuntimeInformation(mv);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/ConfigMethodGen.java:157]: Should we remove this to outside of ConfigMethodGen?

**Comment by HindujaB** [comment]
Created https://github.com/ballerina-platform/ballerina-lang/issues/43214 to address the code coverage issue for runtime API tests.

**Review by gabilang — APPROVED** [review]

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**